rasa/core/migrate.py

```diff
@@ -238,59 +262,114 @@ def _migrate_domain_files(
 
 
 def migrate_domain_format(
-    domain_file: Union[Text, Path], out_file: Union[Text, Path]
+    domain_path: Union[Text, Path], out_path: Optional[Union[Text, Path]],
 ) -> None:
     """Converts 2.0 domain to 3.0 format."""
-    domain_file = Path(domain_file)
-    out_file = Path(out_file)
-    created_out_dir = False
-
-    current_dir = domain_file.parent
+    domain_path = Path(domain_path)
+    out_path = Path(out_path) if out_path else None
+
+    domain_parent_dir = domain_path.parent
+    migrate_file_only = domain_path.is_file()
+
+    # Ensure the backup location does not exist yet
+    # Note: We demand that file as well as folder with this name gets deleted before
+    # the command is run to avoid confusion afterwards.
+    suffix = "original_domain"
+    suffix = f"{suffix}.yml" if migrate_file_only else suffix
+    backup_location = domain_parent_dir / suffix
+    if backup_location.exists():
+        backup_location_str = "directory" if backup_location.isdir() else "file"
+        raise RasaException(
+            f"The domain from '{domain_path}' could not be migrated since the "
+            f"a {backup_location_str} {backup_location} already exists."
+            f"Please remove that there is no file or folder at {backup_location}."
+        )
 
-    if domain_file.is_dir():
-        backup_location = current_dir / "original_domain"
-        backup_location.mkdir()
-
-        if out_file.is_file() or not out_file.exists():
-            out_file = current_dir / "new_domain"
-            out_file.mkdir()
-            created_out_dir = True
-            rasa.shared.utils.io.raise_warning(
-                f"The out path provided is not a directory, "
-                f"creating a new directory '{str(out_file)}' "
-                f"for migrated domain files."
-            )
+    # Choose a default output location if nothing was specified
+    if out_path is None:
+        suffix = DEFAULT_DOMAIN_PATH if migrate_file_only else "new_domain"
```

Comment:
I'm actually debating now if we shouldn't actually use a new suffix for the new domain file, instead of overwriting `DEFAULT_DOMAIN_PATH`, it would be more consistent this way. WDYT?
```suggestion
        suffix = "new_domain.yml" if migrate_file_only else "new_domain"
```

Reply:
yesss, I like that :)